tests/ecmascript/test-bi-object-assign.js

```diff
@@ -0,0 +1,147 @@
+/*
+ *  Object.assign() (E7 19.1.2.1)
+ *  http://www.ecma-international.org/ecma-262/7.0/#sec-object.assign
+ */
+
+/*===
+basic tests
+true
+maggie Kittycow Machel Fatty Whale 812 1208 9001
+0,1,2,pig,cow,ape,whale
+maggie 1208 9001 Vegeta
+cow,pig,ape,saiyan
+undefined
+this belongs here
+undefined
+prop1
+get
+set 812
+===*/
+
+function basicTest() {
+    var obj, src, dest;
+
+    // If no coercion is needed, target object is returned (i.e. not a copy).
+    obj = {};
+    dest = Object.assign(obj, { test: "test" });
+    print(dest === obj);
+
+    // Properties are assigned in [[OwnPropertyKeys]] order, moving from left
+    // to right in the argument list:
+    //     1. Array indices (sequential, ascending)
+    //     2. String keys (creation order)
+    //     3. Symbol keys (creation order)
+    // This behavior is easy to test for by assigning from multiple sources to
+    // an empty object and then calling Reflect.ownKeys() on the result.
```

Comment:
Unfortunately there's no primitive to enumerate stuff in the *internal order*. Which is good in some sense, but unfortunate for white box (well, more "grey box") testing. So getters is all I can think of right now.

To test likely errors well (which is a bit grey box) I think maybe three cases would be needed:
- Object with a mixed set of array / non-array properties, in an order which requires array index sorting. Getters to notice side effects.
- Array with additional properties. All array index properties should be enumerable (and cannot be getters!) to preserve the internal array part. Getters to notice side effects; but these will only be in place for non-array-index properties.
- Array with some array index properties that are non-enumerable. That causes the array part to be abandoned so the internal structure will be different. Insert the non-enumerable orders in a non-ascending order: they will go into the entry part in that order, but then be reordered for OwnPropertyKeys. Again, getters to notice side effects.

I don't think there's a way to test the array-with-array-part case: using a getter for an array causes the array part to be abandoned.